nix/cclib.nix

```diff
@@ -0,0 +1,61 @@
+{ buildPythonPackage
+, lib
+, setuptools
+, pytestCheckHook
+, numpy
+, scipy
+, periodictable
+, pyyaml
+, ase
+, openbabel-bindings
+, h5py
+, pyscf
+, iodata
+, psi4
+, pandas
+, biopython
+, pyquante
+}:
+
+buildPythonPackage rec {
+  pname = "cclib";
+  version = "1.8";
```

Comment:
Honestly this is a little bit of a pain point with Python+Nix. In principle the version tag is mainly used for Nix itself, e.g. what appears at `search.nixos.org` or re-used in the `src` fetcher. But In the case of Python it is also used to specify a version for Setuptools. Say you do a 1.9 release and don't change this line. It will build fine, there won't be any issue except a wrong identifier and possibly some confusion. But as setuptools uses this string, it also needs to be a PEP440 version string, e.g. it cant be something like the git hash or `dev` or whatever. Setuptools will choke on this. But you also can't get it from the git tag or whatever. Both would be an impurity and git directory is removed in Nix builds, usually. I'm not aware of any good solution other than manually adapting this tag, unfortunately. At least it won't break anything :)

Reply:
That is as I suspected.  It isn't in the default branch yet, but I've switched the project to use https://versioningit.readthedocs.io/en/stable/ in a release branch.  This is ok for now; it means the version can no longer be defined purely from a tag.